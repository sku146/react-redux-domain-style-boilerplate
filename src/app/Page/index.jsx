import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

export class Page extends Component {

  constructor(props, context) {
    super(props, context);
  }

  componentDidUpdate({ location }) {
    if (this.props.location !== location) {
      window.scrollTo(0, 0);
    }
  }

  render() {
    return (
      <div className="page-container">
        {this.props.children}
      </div>
    );
  }
}

Page.propTypes = {
  children: PropTypes.node.isRequired,
  location: PropTypes.object.isRequired,
};

export default withRouter(Page);
