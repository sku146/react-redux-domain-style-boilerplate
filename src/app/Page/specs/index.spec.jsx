import React from 'react';
import Page from '../index';

describe('Page', () => {
  const context = {history: mockHistory};
  context.history.push('/');

  const defaultProps = {
    location: context.history.location,
  };

  const RendererComponent = props =>
    shallow(
      <Page {...props}>
        <span>Children</span>
      </Page>,
      {
        lifecycleExperimental: true,
        context
      }
    );

  describe('@render', () => {
    it('should render as expected', () => {
      const component = RendererComponent(defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

  describe('@lifecycle', () => {
    describe('componentDidUpdate()', () => {
      it('If location changes; scroll to 0,0', () => {
        jest.spyOn(global, 'scrollTo');
        const component = RendererComponent(defaultProps);

        component.setProps({location: context.history.location});
        expect(global.scrollTo).not.toHaveBeenCalled();

        context.history.replace('/foo');
        component.setProps({location: context.history.location});
        expect(global.scrollTo).toHaveBeenCalledWith(0, 0);
      });
    });
  });
});
