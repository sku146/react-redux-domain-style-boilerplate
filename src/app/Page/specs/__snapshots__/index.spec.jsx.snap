// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page @render should render as expected 1`] = `<Route />`;
